frontend/packages/dev-console/src/components/helm/HelmChartVersionDropdown.tsx

```diff
@@ -0,0 +1,60 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { safeDump } from 'js-yaml';
+import { FormikValues, useFormikContext } from 'formik';
+import { GridItem } from '@patternfly/react-core';
+import { coFetchJSON } from '@console/internal/co-fetch';
+import { DropdownField } from '@console/shared';
+import { HelmChartMetaData } from './helm-types';
+import { getChartURL, getChartVersions } from './helm-utils';
+
+export type HelmChartVersionDropdownProps = {
+  activeChartVersion: string;
+  chartEntries: HelmChartMetaData[];
+};
+
+const HelmChartVersionDropdown: React.FunctionComponent<HelmChartVersionDropdownProps> = ({
+  activeChartVersion,
+  chartEntries,
+}) => {
+  const { setFieldValue } = useFormikContext<FormikValues>();
+  const [chartVersion, setChartVersion] = React.useState<string>('');
+  const [helmChartVersions, setHelmChartVersions] = React.useState({});
+
+  React.useEffect(() => {
+    const chartVersions = getChartVersions(chartEntries);
+    setHelmChartVersions(chartVersions);
+  }, [chartEntries]);
+
+  const onChartVersionChange = (value: string) => {
+    setChartVersion(value);
+    setFieldValue('chartVersion', value);
+    const url = getChartURL(chartEntries, value);
+    coFetchJSON(`/api/helm/chart?url=${url}`)
+      .then((res) => {
+        setFieldValue('chartValuesYAML', safeDump(res.values));
```

Comment:
Maybe add a safe check here before setting the value similar to what is there while setting initial values in `HelmInstallUpgradePage`.